OneCommunityGlobal/HighestGoodNetworkApp
# Aditya - Fixed Bell Notification for Hours Due in 48 Hours

# Description:
This pull request resolves the bell notification issue for hours due in 48 hours and includes several new features and improvements. 

![image](https://github.com/user-attachments/assets/c81d6a55-01ac-42d4-af5b-81f524f062b8)

## Main changes explained:
- Created a new BellNotification.jsx component to handle the logic for displaying notifications when a user has less than 48 hours remaining to meet their weekly committed hours.
- The notification triggers when the user's logged effort is less than 50% and fewer than 48 hours remain in the week.

## How to test:
1. Check into the current branch.
2. Run npm install.
3. Clear Site Data/Cache: This will ensure no old notifications interfere.
4. Log In: Log in as any user.
5. Check the Notification Bell: Navigate to the Dashboard and verify the notification bell functionality.
6. Test Time Manipulation: Use the "Add Intangible Hours" function in the Timelog to manipulate the logged time.
7. Trigger Notification: Verify that the bell notification triggers when less than 50% of hours are logged and only 48 hours remain.
8. Test Reset: Confirm the notification resets after it is viewed or when the week ends.
9. Test Dark Mode: Ensure the notification popup works properly in both light and dark modes.

## Screenshots or videos of changes:

https://github.com/user-attachments/assets/64a1d65e-1c16-4459-9a0c-9c993bdf2505

Add Intangible Time Entry Button:

![image](https://github.com/user-attachments/assets/f6c7b8d9-1919-466b-99f0-826f3a27a4a7)

Press the Button and add/edit/delete Time:

![image](https://github.com/user-attachments/assets/7361a11e-80c1-4196-8d7f-0c3f72e1fed3)

## Note:
1. Modify Time Left: To test different time thresholds for notifications, modify the hoursLeft value in the BellNotification.jsx component.
![image](https://github.com/user-attachments/assets/89be01d2-551a-4740-98ad-3429205cd784)
2. Simulate Time Logs: Use the "Add Intangible Hours" feature to simulate different time log scenarios.
3. Clear Cookies for Retesting: The notification triggers only once per week. Clear cookies or localStorage to reset the notification for retesting.
